lnrpc/autopilotrpc/autopilot_server.go

```diff
@@ -207,3 +214,31 @@ func (s *Server) QueryScores(ctx context.Context, in *QueryScoresRequest) (
 
 	return resp, nil
 }
+
+// SetScores sets the scores of the external score heuristic, if active.
+//
+// NOTE: Part of the AutopilotServer interface.
+func (s *Server) SetScores(ctx context.Context,
+	in *SetScoresRequest) (*SetScoresResponse, error) {
+
+	scores := make(map[autopilot.NodeID]float64)
+	for pubStr, score := range in.Scores {
+		pubHex, err := hex.DecodeString(pubStr)
+		if err != nil {
+			return nil, err
+		}
+		pubKey, err := btcec.ParsePubKey(pubHex, btcec.S256())
+		if err != nil {
+			return nil, err
+		}
+		nID := autopilot.NewNodeID(pubKey)
+
+		scores[nID] = float64(score)
```

Comment:
score is an int, so this only supports scores of 0 and 1?

Reply:
Fixed: https://github.com/lightningnetwork/lnd/pull/2212/commits/2b70f6d1245ed3175d29eba248797def8cdb1bf7